### PR TITLE
Support tiled Copernicus DEM 30m

### PR DESCRIPTION
**Pull request by uchchwhash**
This is a step towards processing ARD globally.

The `cop_pathname` setting in luigi config now additionally supports an s3 bucket address or a folder containing the tiled Copernicus DEM.

Thanks @mwheeler for doing the research work.

### CODE REVIEW

**Comment by sixy6e** [comment]
What's the status of this PR?  When is it likely to be merged?

@truth-quark, we'll need this PR for the Antarctic work.

**Comment by uchchwhash** [comment]
Just got the green light from Fuqin Li. Please review @jeremyh @sixy6e.

File: wagl/dsm.py [hunk 1]
```diff
@@ -42,6 +50,183 @@ def filter_dsm(array):
...
+            try:
+                reprojected = np.full(subset_geobox.shape, np.nan, dtype=np.float32)
+
+                reproject(
+                    source=rasterio.band(ds, 1),
+                    destination=reprojected,
+                    dst_crs=dst_crs,
+                    dst_transform=subset_geobox.transform,
+                    dst_nodata=np.nan,
+                    resampling=Resampling.bilinear,
```
Comment (sixy6e) [inline wagl/dsm.py:217]: This may need tweaking in future if a different resampling method is desired. But good for now.

**Review by sixy6e — APPROVED** [review]
I'm happy with the content. Might need a little evolving in future, and probably something that could be done from the antarctic project work.

**Pull request merged**